internal/controllers/s3bucket/controller.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package s3bucket
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	s3v1alpha1 "github.com/snapp-incubator/s3-operator/api/v1alpha1"
+)
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&s3v1alpha1.S3Bucket{}).
+		// Set predicate to filter only generation change events.
```

Comment:
What's the reasoning behind using this predicate?

I was thinking about those times when we want to force schedule a reconciliation request by adding a simple `a=b` label. This predicate will prevent those requests I think.

Reply:
When the reconciler updates the status in the creating process (provision), it will create another update event, which triggers another reconciliation, hence a redundant provision. I saw in the logs that Ceph's requests are twice. So, to prevent such behavior, I followed the practice [here](https://cloud.redhat.com/blog/kubernetes-operators-best-practices) in the _managin status_ section.
The same happens for adding or deleting the finalizer, and that predicate avoids it.

Reply:
Thanks for your clarification, dear Hamed.